src/Cake.Common/Tools/MSTest/MSTestRunner.cs

```diff
@@ -114,12 +114,31 @@ protected override IEnumerable<string> GetToolExecutableNames()
         /// <returns>The default tool path.</returns>
         protected override IEnumerable<FilePath> GetAlternativeToolPaths(MSTestSettings settings)
         {
-            foreach (var yearAndEdition in new[] { "2019/Enterprise", "2019/Professional", "2019/Community", "2017/Enterprise", "2017/Professional", "2017/Community" })
+            foreach (var year in new[] { "2022", "2019", "2017" })
             {
-                var path = GetYearAndEditionToolPath(yearAndEdition);
-                if (_fileSystem.Exist(path))
+                var editions = new List<string>(
+                    new[]
+                    {
+                        "Enterprise",
+                        "Professional",
+                        "Community",
+                        "BuildTools",
+                    });
+
+                if (settings.AllowPreviewVersion)
                 {
-                    yield return path;
+                    editions.Insert(0, "Preview");
```

Comment:
Same as above... Shouldn't `Preview` be the last option / fallback if none of the other editions were found? 🤔 

```suggestion
                    editions.Add("Preview");
```

Reply:
See my comment in https://github.com/cake-build/cake/pull/3775#discussion_r772449501